### PR TITLE
Optional aws upload

### LINKED ISSUE #42: Add AWS upload option to config
### What can be improved?

At the moment, AWS docker image upload workflow jobs are active by default (and do nothing if no AWS secrets are defined). This can be confusing to non-CML users who do not interact with AWS _at all_.

The easiest would be to scrub all mention of AWS if the user is not interested in uploading a docker image of the package there (much like we do with pypi and conda uploads).

### Version

v0.2.0

### PR DESCRIPTION
**Pull request by brynpickering**
Fixes #42

### CODE REVIEW

File: schema.yaml [hunk 1]
```diff
@@ -80,6 +80,15 @@ properties:
...
     items: *n_y_items
 
+  upload_aws_image:
+    type: array
+    description: >-
+      Whether to upload the repository to AWS to build and host a Docker image of the project.
+      NOTE: requires CodeBuild configuration to be updated separately to have the image built on AWS.
+    uniqueItems: true
+    minItems: 1
+    items: *n_y_items
```
Comment (mfitz) [inline schema.yaml:90]: I see some of the other y/n params have additional config like this, where the default and options are explicit:

```
      default: "n"
      enum: ["y", "n"]
```

Is there an implicit default here? Based on the unit tests, it seems to default to "no". Is that correct?

File: {{cookiecutter.repository_name}}/.github/workflows/commit-ci.yml [hunk 2]
```diff
@@ -23,6 +23,7 @@ jobs:
       {%- endif %}
       lint: false
 
+  {%- if cookiecutter.upload_aws_image|lower == "y" %}
```
Comment (mfitz) [inline {{cookiecutter.repository_name}}/.github/workflows/commit-ci.yml:26]: Ah, this answers the question about the default value...

**Review by mfitz — APPROVED** [review]
Looks good.

I had a question, but it was answered when I scrolled down 😄

File: schema.yaml (hunk 1 above)
Comment (brynpickering) [reply schema.yaml:90]: `*n_y_items` is a YAML anchor call to `&n_y_items`, in which those values are defined.

**Pull request merged**